app/assets/javascripts/lookup_keys.js

```diff
@@ -53,7 +53,7 @@ function fix_template_context(content, context) {
 
 function fix_template_names(content, assoc, new_id) {
   var regexp  = new RegExp('new_' + assoc, 'g');
-  return content.replace(regexp, "new_" + new_id);
+  return content.replace(regexp, new_id);
 }
```

Comment:
Yes.  When strong_attribute filters nested parameters (in the form `:host => {:host_parameters_attributes => {'1' => {:name => 'foo', :value => 'bar'}}`), then you call `params.require(:host).permit(:host_parameters_attributes => [[:name, :value]])` to indicate that it's an array of hashes with the keys name, value.

Even though the input is a hash (as generated by the Rails nested forms), strong parameters understands it's essentially an array, but _only_ if all of the keys are numeric.  If there are any non-numeric keys then the entire sub-hash is filtered out.

The change here ensures that when a new child object is created in our forms (e.g. a new NIC or parameter), that it has a solely numeric key and not "new_123456789".  The changes to serializeForm() are similar and filter out the child form templates - if the form templates are included then we end up with numeric keys for all of the existing and new nested items, but also a "new_interface" item that causes the whole hash to be filtered.